### PR TITLE
Add variant_name kwarg

### PR DESCRIPTION
**Pull request by hardianlawi**
*Issue #, if available:*

https://github.com/aws/sagemaker-python-sdk/issues/1122

*Description of changes:*

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [ ] I certify that the changes I am introducing will be backword compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [ ] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by vikas0203 — APPROVED** [review]

**Review by vikas0203 — COMMENTED** [review]
/bot run all

**Review by vikas0203 — COMMENTED** [review]
/bot run slow-tests

**Review by vikas0203 — COMMENTED** [review]
/bot run unit-tests

**Review by vikas0203 — COMMENTED** [review]
/bot run all

**Review by vikas0203 — COMMENTED** [review]
/bot run slow-tests,unit-tests

**Review by vikas0203 — COMMENTED** [review]
/bot run all

**Review by vikas0203 — COMMENTED** [review]
/bot run slow-tests,unit-tests

**Review by vikas0203 — COMMENTED** [review]
/bot run slow-tests,unit-tests

**Review by vikas0203 — COMMENTED** [review]
/bot run slow-tests,unit-tests

**Review by vikas0203 — COMMENTED** [review]
/bot run unit-tests

**Comment by npow** [comment]
Hi, any idea when this will be merged in master? It's been open for ~6 months now.

**Review by vikas0203 — COMMENTED** [review]
/bot run unit-tests

**Review by vikas0203 — COMMENTED** [review]
/bot run unit-tests

File: src/sagemaker/huggingface/model.py [hunk 1]
```diff
@@ -200,6 +200,7 @@ def deploy(
         deserializer=None,
         accelerator_type=None,
         endpoint_name=None,
+        variant_name="AllTraffic",
```
Comment (mufaddal-rohawala) [inline src/sagemaker/huggingface/model.py:203]: Since this is a new argument, can we add this at the end of argument list?

**Comment by mufaddal-rohawala** [comment]
Apologies for late comment, but we will be taking of tracking this PR to completion.

**Review by mufaddal-rohawala — COMMENTED** [review]
/bot run all

**Review by mufaddal-rohawala — COMMENTED** [review]
/bot run all

**Comment by mufaddal-rohawala** [comment]
Due to low response on addressing the PR comments, we are going to close this pull-request. We will keep the associated  issue open to track and fix using this PR for referrence.

**Pull request closed without merging**